browser/src/docstatefunctions.js

```diff
@@ -150,3 +150,15 @@ app.isExperimentalMode = function () {
 		return app.socket.WSDServer.Options.indexOf('E') !== -1;
 	return false;
 };
+
+app.getUnsavedCloseMessage = function () {
+	return _('You have unsaved changes. Are you sure you want to quit?');
+};
+
+app.preventExit = function () {
+	window.onbeforeunload = app.getUnsavedCloseMessage;
+};
+
```

Comment:
what about just using true/false ? :-)

Reply:
`Setting the event object's [returnValue](https://developer.mozilla.org/en-US/docs/Web/API/BeforeUnloadEvent/returnValue) property to a non-empty string value or any other [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) value.`

String can be used by some browsers to be shown in the popup.